pkg/build/registry/buildlog/rest.go

```diff
@@ -52,16 +52,16 @@ func (r *REST) ResourceLocation(ctx kapi.Context, id string) (string, error) {
 
 	// TODO: these must be status errors, not field errors
 	// TODO: choose a more appropriate "try again later" status code, like 202
-	if len(build.PodName) == 0 {
-		return "", errors.NewFieldRequired("Build.PodName", build.PodName)
+	if build.PodRef == nil || len(build.PodRef.Name) == 0 || len(build.PodRef.Namespace) == 0 {
+		return "", errors.NewFieldRequired("Build.PodRef", build.PodRef)
```

Comment:
Should this be a `Validate...` call instead? Also, I think the validation path should  be `Build.podRef` (kinds are uppercased, fields are lower)